rotkehlchen/chain/evm/decoding/monerium/decoder.py

```diff
@@ -0,0 +1,110 @@
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.monerium.constants import MIGRATED
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+from .constants import CPT_MONERIUM
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class MoneriumCommonDecoder(DecoderInterface):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            native_asset: Asset,
+            asset_address: ChecksumEvmAddress,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.native_asset = native_asset
+        self.asset_address = asset_address
+
+    def _decode_monerium(self, context: DecoderContext) -> DecodingOutput:
+        """Decode Monerium when it is minted or burned"""
+
+        if context.tx_log.topics[0] != MIGRATED:
+            return DEFAULT_DECODING_OUTPUT
+
+        from_address = hex_or_bytes_to_address(context.all_logs[0].topics[1])
+        to_address = hex_or_bytes_to_address(context.all_logs[0].topics[2])
+
+        if not self.base.is_tracked(from_address) and not self.base.is_tracked(to_address):
+            return DEFAULT_DECODING_OUTPUT
+
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data)
+        amount = token_normalized_value_decimals(amount_raw, 18)
+
+        event = None
+        if from_address == '0x0000000000000000000000000000000000000000':
+            # Create a mint event
+            mint_event = self.base.make_event_from_transaction(
+                transaction=context.transaction,
+                tx_log=context.tx_log,
+                event_type=HistoryEventType.RECEIVE,
+                event_subtype=HistoryEventSubType.NONE,
+                asset=self.native_asset,
+                balance=Balance(amount=amount),
+                location_label=to_address,
+                notes=f'Mint {amount} EURe',
+                counterparty=CPT_MONERIUM,
+                address=self.asset_address,
+            )
+            event = mint_event
```

Comment:
technically this is correct but presents a problem.

The reason for you to use `make_event_from_transaction` instead of reading the events of `context.decoded_events` is because you are adding a decoding rule for

```
token_contract -> this function
```

and this overwrites any other decoding logic skipping the default processing that we do in https://github.com/rotki/rotki/blob/97d07a154ec7dfdd480fb10e406ce2c9aa101bb8/rotkehlchen/chain/evm/decoding/decoder.py#L809.

This presents a problem and it is that if you have two addresses `addy_a` and `addy_b` and you make a transfer
`addy_a` -> `addy_b` this function that you created will process the transfer and since is not from the ZERO_ADDRESS or to the ZERO_ADDRESS it won't get decoded and it won't appear in the list of events.

To overcome this issue instead of address_to_decoders use 

```
    def decoding_by_input_data(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
        return {
            b'D\x9aR\xf8': {  # mint event
                ERC20_OR_ERC721_TRANSFER: self._decode_monerium_mint_and_burn,
            },
        }
```

and also add the burn event. Note that since this logic will now only receive mint/burn events you might be able to remove some checks.

Before doing any change test this transaction https://etherscan.io/tx/0xf92d5493059ff3a4dc30039638c521b16cdb867618fe226f6f879cbe97e96020 that should produce no event if you track any of the addresses and then with the change I mentioned it should work properly

Reply:
as you were using address_to_decoders we execute this logic in this case

```
            if decoding_output.event:
                events.append(decoding_output.event)
                continue
```

and the execution breaks not processing any of the other common rules that we have